src/pages/billing/NewBillingProjectWizard/AzureBillingProjectWizard/AzureBillingProjectWizard.ts

```diff
@@ -0,0 +1,160 @@
+  // undefined used to indicate that the user has not yet typed in the input (don't want to show error)
+  const [billingProjectName, setBillingProjectName] = useState<string|undefined>(undefined)
+  const [userEmails, setUserEmails] = useState({ emails: '', hasError: false })
+  const [ownerEmails, setOwnerEmails] = useState({ emails: '', hasError: false })
+  // undefined used to indicate that the user has not yet made a selection
+  const [addUsersOrOwners, setAddUsersOrOwners] = useState<boolean|undefined>(undefined)
+  const [managedApp, setManagedApp] = useState<AzureManagedAppCoordinates>()
+
+  const [existingProjectNames, setExistingProjectNames] = useState<string[]>([])
+  const [projectNameErrors, setProjectNameErrors] = useState<ReactNode>()
+
+  const [isBusy, setIsBusy] = useState(false)
+
+  const createBillingProject = _.flow(
+    withBusyState(setIsBusy),
+    reportErrorAndRethrow('Error creating billing project'),
+  )(async () => {
+    if (!billingProjectName) return
+    try {
+      const users = userInfoListToProjectAccessObjects(userEmails.emails, 'User')
+      const owners = userInfoListToProjectAccessObjects(ownerEmails.emails, 'Owner')
+      const members = _.concat(users, owners)
+      await Ajax().Billing
+        .createAzureProject(
+          billingProjectName, managedApp?.tenantId, subscriptionId, managedApp?.managedResourceGroupId, members
+        )
+      onSuccess(billingProjectName)
+      // No need to event success, as that is done in the onSuccess callback
+    } catch (error: any) {
+      if (error?.status === 409) {
+        setExistingProjectNames(_.concat(billingProjectName, existingProjectNames))
+        Ajax().Metrics.captureEvent(Events.billingAzureCreationProjectDuplicateName)
+      } else {
+        Ajax().Metrics.captureEvent(Events.billingAzureCreationProjectCreateFail)
+        throw error
+      }
+    }
+  })
+
+  useEffect(() => {
+    const errors = Utils.cond(
+      [!!billingProjectName, () => summarizeErrors(
+        validate({ billingProjectName }, { billingProjectName: billingProjectNameValidator(existingProjectNames) })?.billingProjectName
+      )],
+      [billingProjectName !== undefined, () => 'A name is required to create a billing project.'],
+      [Utils.DEFAULT, () => undefined]
+    )
+    setProjectNameErrors(errors)
+  }, [billingProjectName, existingProjectNames])
+
+  const stepFinished = (step: number, finished: boolean) => {
+    if (finished && activeStep === step) { // the user completed the active step
+      setActiveStep(step + 1)
+    } else if (!finished && activeStep > step) { // the user went back
+      setActiveStep(step)
+    } // the user is entering fields for later steps - don't change active step
+  }
+
+  const onManagedAppSelected = managedApp => {
+    stepFinished(1, !!managedApp)
+    setManagedApp(managedApp)
+  }
+
+  const step1HasNoErrors = !!subscriptionId && !!managedApp
```

Comment:
This step transition logic isn't overly tested-- just the happy path. I ran out of time to figure out how to write more thorough tests.